integration-tests/src/test/resources/field_level_security_mode/readonlyrest.yml

```diff
@@ -0,0 +1,32 @@
+readonlyrest:
+  ssl:
+    enable: true
+    keystore_file: "keystore.jks"
+    keystore_pass: readonlyrest
+    key_pass: readonlyrest
+
+  access_control_rules:
+
+    # ES containter initializer need this rule to configure ES instance after startup
+    - name: "CONTAINER ADMIN"
+      auth_key: admin:container
+      verbosity: error
+
+    - name: fls_mode_default
+      auth_key: "user1:pass"
+      fields: ["_source","allowedField"]
+
+    - name: fls_mode_hybrid_explicit
+      auth_key: "user2:pass"
+      fields: ["_source","allowedField"]
+      fields_mode: "hybrid"
+
+    - name: fls_mode_legacy_explicit
+      auth_key: "user3:pass"
+      fields: ["_source","allowedField"]
+      fields_mode: "legacy"
```

Comment:
we should consider mode names changes. I don't know if "legacy" and "hybrid" are good names from perspective of client. From developer perspective they obviously make sense

Reply:
Please remember we are in he business of removing decision points from the user experience, not adding them. From the user point of view, what would functionally change between hybrid and legacy?

Reply:
we were not managed to improve `fields` rule as we planned to do at the beginning. So, here we have a hybrid approach. The hybrid approach still requires to install ROR on all nodes. But we give the user an option to pick - the fast `fields` rule with ROR on one node (but for some requests ROR will return `forbidden`) or pretty fast `fields` rule with ROR on all nodes (all requests will be handled). The hybrid will be a default, but the user can pick the second one instead - depending on his requirements.

Obviously, the configuration above has no sense - we've discussed it and agreed that a global setting will be better. You will see it in the next commit

Reply:
OK for the default=hybrid, and people will just observe a speed up in a good most commonly used set of filtered requests. Instead I don't feel the case for "legacy" mode, as if somebody requires to set it, there must be a bug in the new hybrid solution. Instead for the "proxy" mode I'd say more pre-routing or pre-shard, as this enhancement is not only for proxy.

Reply:
we have 3 options which will be configured globally, but only 2 of them will be mentioned in doc:
1. legacy mode (works as before) - when there will be a bug and the customer will complain about it, we can instruct him to enable this mode (it will be removed in future, when the solution warms up).
2. hybrid mode (default = es level fields handling + lucene level fields handling) - admin have to install ROR on each ES node + all requests will be handled
3. es level fields handling only (all requests which cannot be handled like this will be rejected) - admin can install ROR only on one node

So, by default you will have 2nd mode enabled - a fast and complete solution. If you don't want to install ROR on each node and you accept limitations of 3rd mode, you can set it up.